sql/src/main/java/io/crate/analyze/where/WhereClauseNullReplacer.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.where;
```

Comment:
I wonder if we should change the package structure here a bit more as well. E.g.

`io.crate.execution.expression -> io.crate.expression` (maybe `.eval`)
`io.crate.analyze.symbol -> io.crate.expression.symbol`

To group everything related to expression evaluation closer together and making it more independent of *where* it is used.
(Maybe in a separate follow up PR, to make backporting less of a hassle?)